core/vulkan/loader/loader.go

```diff
@@ -16,40 +16,127 @@
 package loader
 
 import (
+	"bytes"
 	"context"
 	"io/ioutil"
 	"os"
 	"runtime"
 	"strings"
 
 	"github.com/google/gapid/core/app/layout"
+	"github.com/google/gapid/core/os/device"
+	"github.com/google/gapid/core/os/device/bind"
+	"github.com/google/gapid/core/os/device/remotessh"
 	"github.com/google/gapid/core/os/file"
 	"github.com/google/gapid/core/os/shell"
 )
 
+// DeviceSetup handles getting files from/to the right location
+// for a particular Device
+type DeviceReplaySetup interface {
+	// MakeTempDir returns a path to a created temporary
+	MakeTempDir(ctx context.Context) (string, func(ctx context.Context), error)
```

Comment:
Please document the `func(ctx context.Context)` (which almost has the `task.Task` signature)

Reply:
Done